scripts/tools/pad_number_generators.py

```diff
@@ -23,9 +24,22 @@
 
 def increment(pincount, init=1, **kwargs):
     i = init
+    j = init # pad number for deleted pins
+
+    if kwargs.get("deleted_pins"):
+        skip_pins = kwargs["deleted_pins"]
```

Comment:
I really don't like this naming. I think the variable should be called `deleted_pins` as well.

Reply:
I tried to abstract things. In the PadArray node, for example, I kept the exclude_pin_list variable name but I allow two entry points of hidden_pins and deleted_pins since there are some cases where the same list works and other places where I need to know if the pins are hidden or deleted. Similarly, I decided to use a general term under the hood that could be used for other purposes in the future while the parameter name is specific to the behavior the user requests. .I can change it if you like.

Reply:
How about that:
```
def increment(pincount, init=1, skip_pins=[]):
```
That makes it more abstract, it does not need to know about the `deleted_pins` keyword.

Reply:
I thought about that, but I will still have to pick this keyword up in `get_generator()` so I figured it was essentially the same thing one place or the other. Is it better to grab this keyword in one place or the other?

Reply:
I think it is better to extract the keyword in the calling function.
That way the `increment` is more abstract. Also the user/programmer can clearly see what it does when looking at the arguments.

Reply:
Yes, that's what I was feeling. Sorry for not being more clear and/or verbose. I originally had the pin-skipping increment generator as a separate function but decided from the outside it was similar enough to just combine them. I'll leave this alone unless I hear another comment from you.

Reply:
Currently, custom pad numbering doesn't work with deleted pins. If that's going to work in the future the refactoring will need to be done since the pins to be deleted will have to be removed for any generator function and not just the increment one. But that can happen when/if needed.